mentoring/answer.py

```diff
@@ -70,43 +174,39 @@ def student_input(self):
 
         return student_input
 
-    def mentoring_view(self, context=None):
-        if not self.read_only:
-            html = render_template('templates/html/answer_editable.html', {
-                'self': self,
-            })
-        else:
-            html = render_template('templates/html/answer_read_only.html', {
-                'self': self,
-            })
-        
-        fragment = Fragment(html)
-        fragment.add_css_url(self.runtime.local_resource_url(self.xblock_container, 'public/css/answer.css'))
-        fragment.add_javascript_url(self.runtime.local_resource_url(self.xblock_container,
-                                                                    'public/js/answer.js'))
-        fragment.initialize_js('AnswerBlock')
-        return fragment
+    def fallback_view(self, view_name, context=None):
+        context = context or {}
+        context['self'] = self
+        html = loader.render_template('templates/html/answer_editable.html', context)
 
-    def mentoring_table_view(self, context=None):
-        html = render_template('templates/html/answer_table.html', {
-            'self': self,
-        })
         fragment = Fragment(html)
-        fragment.add_css_url(self.runtime.local_resource_url(self.xblock_container, 'public/css/answer_table.css'))
+        fragment.add_css_url(self.runtime.local_resource_url(self, 'public/css/answer.css'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/answer.js'))
+        fragment.initialize_js('AnswerBlock')
         return fragment
 
     def submit(self, submission):
-        if not self.read_only:
-            self.student_input = submission[0]['value'].strip()
-            log.info(u'Answer submitted for`{}`: "{}"'.format(self.name, self.student_input))
+        self.student_input = submission[0]['value'].strip()
+        self.save()
```

Comment:
Good question. The main reason is that, despite appearances, this is not a formal XBlock handler method (notice it has no decorator). Instead this is just a method on the XBlock that gets called by the parent block (mentoring), which is the one that has the real "submit" handler and manages the "submit" process on behalf of all of its child blocks. So the runtime doesn't know that this block wants to persist any changes unless we explicitly call save.

Also, note that this XBlock overrides the `save()` method, in order to save its data into a Django model.